src/js/wall/B2cLogin.js

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import { connect } from "react-redux";
+import styled from "styled-components";
+import { loginSucceeded } from "../account/actions";
+import { getLoginRequest } from "../app/authConfig";
+import { getColor } from "../app/theme";
+import { Button, Divider } from "../base";
+import { WallSubheader } from "./Container";
+
+const StyledB2cLogin = styled.div`
+    display: flex;
+    flex-direction: column;
+    align-items: stretch;
+    margin: 10px 0px;
+    padding: 0px 15px;
+`;
+
+const BetaTag = styled.span`
+    color: ${props => getColor({ color: "greyDark", theme: props.theme })};
+    margin-left: auto;
+    transform-text: uppercase;
+`;
+
+const StyledDivider = styled(Divider)`
+    margin-bottom: 30px;
+`;
+
+export const B2cLogin = ({ setLogin }) => (
+    <>
+        <StyledB2cLogin>
+            <WallSubheader>Sign in with your work account</WallSubheader>
+            <Button onClick={() => window.msalInstance.loginPopup(getLoginRequest()).then(setLogin)}>Sign in</Button>
+            <BetaTag>Beta</BetaTag>
+        </StyledB2cLogin>
+        <StyledDivider text="or" />
+    </>
+);
+
+export const mapDispatchToProps = dispatch => ({
+    setLogin: msal => {
+        window.msalInstance.setActiveAccount(msal.account);
+        dispatch(loginSucceeded());
+    }
```

Comment:
I don't think this is not the appropriate place for this.

Only the call to `dispatch` should be here. I would just define an `onClick` the deals with all the `msal` business and finishes up with a call to the `setLogin` prop.